zulipterminal/ui_tools/views.py

```diff
@@ -589,3 +590,107 @@ def keypress(self, size: Tuple[int, int], key: str) -> str:
         if is_command_key('GO_BACK', key) or is_command_key('HELP', key):
             self.controller.exit_pop_up()
         return super(HelpView, self).keypress(size, key)
+
+
+class MsgInfoView(urwid.ListBox):
+    def __init__(self, controller: Any, msg: Any) -> None:
+        self.controller = controller
+        self.msg = msg
+
+        if msg['type'] == 'stream':
+            recipient = msg['display_recipient'] + " -> " + msg['subject']
+        elif not msg['display_recipient'][0]['full_name'] ==\
+                msg['sender_full_name']:
+            recipient = msg['display_recipient'][0]['full_name']
+        else:
+            recipient = 'Self'
+
+        msg_info = OrderedDict({
+            'Date & Time': time.ctime(msg['timestamp'])[:-5],
+            'Sender': msg['sender_full_name'],
+            'Sender\'s Email ID': msg['sender_email'],
+            'Recipient': recipient,
+            'Reactions': sorted(
+                    [reaction['emoji_name'] +
+                        ": " +
+                        reaction['user']['full_name'] +
+                        "\n"
+                        for reaction in msg['reactions']])
+            if msg['reactions']
+            else '---None---',
+            'Edit History': 'Press \'e\' to view',
+            })
+
+        widths = [(len(field[0])+7,
+                  max(len(reaction_users) for reaction_users in field[1])
+                  if isinstance(field[1], list)
+                  else len(field[1])+2)
+                  for field in msg_info.items()]
```

Comment:
I'd again suggest using a tuple in this for loop rather than indexing `field`, ie. using the `for key, value in some_dict.items()` style.